ledger_device_sdk/src/lib.rs

```diff
@@ -52,6 +53,43 @@ macro_rules! set_panic {
     };
 }
 
+#[cfg(feature = "heap")]
+use critical_section::RawRestoreState;
+#[cfg(feature = "heap")]
+use embedded_alloc::Heap;
+
+#[cfg(feature = "heap")]
+#[global_allocator]
+static HEAP: Heap = Heap::empty();
+
+#[cfg(feature = "heap")]
+struct CriticalSection;
+#[cfg(feature = "heap")]
+critical_section::set_impl!(CriticalSection);
+
+/// Default empty implementation as we don't have concurrency.
+#[cfg(feature = "heap")]
+unsafe impl critical_section::Impl for CriticalSection {
+    unsafe fn acquire() -> RawRestoreState {}
+    unsafe fn release(restore_state: RawRestoreState) {}
+}
+
+/// Initializes the heap memory for the global allocator.
+///
+/// The heap is stored in the stack, and has a fixed size.
+/// This method is called just before [sample_main].
+#[no_mangle]
+#[cfg(feature = "heap")]
+extern "C" fn heap_init() {
```

Comment:
Question: as it looks like `heap_init()`is only used by the sys crate, why not defining this function in the sys crate as well ?

Reply:
Good suggestion. I moved all the heap code to `ledger_secure_sdk_sys`.